### ISSUE TITLE
bug: access-control-allow-origin should be set to localhost

### ISSUE THREAD

**Issue by weboko** [waku-org/nwaku#2223]
### Problem
Cannot read data returned from REST API in browser if request is done with CORS violation.

### Impact
Prevents browsers from querying REST API.

### To reproduce
0. Run nwaku REST API
1. Copy https://github.com/waku-org/waku-frontend/tree/weboko/follow-up
2. Run `npm i`
3. Run `npm run dev`
4. Open `localhost:3000`
5. Wait for 10 seconds and see in console following error:
![image](https://github.com/waku-org/nwaku/assets/118575614/356ac598-5098-4e00-bde2-83a65ddb8b09)

It happens even though the request succeds. 
![image](https://github.com/waku-org/nwaku/assets/118575614/d191758b-373e-443d-a892-f70d2aa11d84)

### Expected behavior
Error should not happen.

### Additional context
What happens is that browser implements Fetch API in the manner that when request is made to resource with CORS violation then even if it would succeed - client won't be able to read response data. 
Spec of the Fetch API - https://fetch.spec.whatwg.org/#concept-filtered-response-opaque

To clarify why even if request is made from the same host (in our case localhost) it is considered violation - `Two URLs have the same origin if the protocol, port(if specified), and host are the same for both.`, [source](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)

### The fix
Considering we expect REST API to be run only on localhost we should add following HTTP header to allow web apps run on different port to be able to talk to the API.
```
Access-Control-Allow-Origin: http://localhost:3000, http://localhost:80, http://localhost:8080, https://localhost:3000, https://localhost:80, https://localhost:8080
```
or 
```
Access-Control-Allow-Origin: *
```

**Comment by NagyZoltanPeter** [comment]
With @weboko found the root cause of the issue and a possible solution for it as well!
It is a limitation in nim-presto lib implementation of HttpServer. It can only match with exact and exactly one allowedOrigin if that is not a single '*'.

The solution is to extend originMatch function at https://github.com/waku-org/nwaku/blob/5ca16485e4d74e531d50d289ebc0f869d9e6352b/presto/serverprivate.nim#L44-L53

It should handle comma separated list of allowedOrigins and be able to match with wildcarded ports definition.

https://github.com/waku-org/nwaku/blob/846f7505ba20b19d582f8fd74e784af37147a25b/waku/waku_api/rest/server.nim#L96
Example: 
1. "http://localhost:*,http://127.0.0.1:*"

this one should successfully matched against: origin[0]="http://localhost:3000"

**Comment by gabrielmer** [comment]
**Weekly Update**

- _achieved_: understood and applied initial fixes, tested, found bug and got it to work
- _next_: add tests for changes in presto, raise PRs in both presto and nwaku

**Comment by NagyZoltanPeter** [comment]
Need to wait for completion for PR to be reviewed on nim-presto: https://github.com/status-im/nim-presto/pull/68

**Comment by NagyZoltanPeter** [comment]
With Eugen we discussed it through.
The PR https://github.com/status-im/nim-presto/pull/68 will not be accpted as it introduces unwanted dependency to regex library.

Also in the meanwhile there are certain architectural development. First it happaned on nim-chronos with HttpServer.
It adds a great option to the lib users to influence the request processing by adding a chain of callbacks (aka middlewares). 
Last week I did several design trials if that alone is capable to implement CORS headers handling inside nim-waku.
Unfortunately there are design constraints on nim-presto RestHttpServer and Router implementation that prevents this to easily happen and I wanted to avoid to import large portion of code from nim-presto into nim-waku just for this sole feature. Also it would have had bad impact for future code maintenance.

Thus we agreed on an improvement on nim-presto library and the RestHttpServer itself that will alow us to add such a feature more easy without compromising design and code quality.
Therefore to implement this fix/chore for nimwaku's rest service we need to wait for that implementation on nim-presto. Estimated to get it by 24-25 of  January.
cc: @chair28980
